standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1409,6 +1407,23 @@ public void commitTxn(CommitTxnRequest rqst)
     }
   }
 
+  /**
+   * Create Notifiaction Events on txn commit
+   * @param dbConn DatabaseConnection
+   * @param txnid committed txn
+   * @param txnType transaction type
+   * @throws MetaException ex
+   */
+  protected void createCommitNotificationEvent(Connection dbConn, long txnid, Optional<TxnType> txnType)
+      throws MetaException, SQLException {
+    if (transactionalListeners != null) {
+      MetaStoreListenerNotifier.notifyEventWithDirectSql(transactionalListeners,
+              EventMessage.EventType.COMMIT_TXN, new CommitTxnEvent(txnid, txnType.get()), dbConn, sqlGenerator);
+    }
+  }
+
+  protected abstract CompactionInfo getCompactionByTxnId(Connection dbConn, long txnid) throws SQLException, MetaException;
```

Comment:
why do we have compaction related stuff in TxnHandler?

Reply:
It was a leftover, deleted.